### PR TITLE
Gazebo Simulator Work

### PR DESCRIPTION
**Pull request by mattlangford**
Namely, added transdec model, put some field elements in some location that is close to their actual positions, and improved the simulator back end. Things to do in the near future:
1.  Make sure models have correct dimensions.
2.  Put all field elements in the correct locations.
3.  Add torpedo and gripper onto the sub's model.

### CODE REVIEW

File: legacy/rise_6dof/scripts/rise_6dof [hunk 1]
```diff
@@ -65,8 +65,8 @@ class Node(object):
         self.constant_wrench_time = None
         rospy.Service('send_constant_wrench', SendConstantWrench, self.send_constant_wrench)
 
-        pub = rospy.Publisher('/output', WrenchStamped)
-        pd_pub = rospy.Publisher('/pd_output', WrenchStamped)
+        pub = rospy.Publisher('/output', WrenchStamped, queue_size=2)
```
Comment (jpanikulam) [inline legacy/rise_6dof/scripts/rise_6dof:68]: Nice

File: simulation/sub8_gazebo/models/sub8/sub8.sdf [hunk 2]
```diff
@@ -145,6 +145,34 @@
           <Cy>240</Cy>
         </plugin>
       </sensor>
+
+      <!-- Uses LIDAR plugin to simulate 1 LIDAR ray to get distance to floor.
+           Best solution would be to write custom plugin but fuck it.-->
```
Comment (jpanikulam) [inline simulation/sub8_gazebo/models/sub8/sub8.sdf:150]: :)

File: simulation/sub8_gazebo/nodes/gazebo_controller.py [hunk 3]
```diff
@@ -50,12 +53,13 @@ def wrench_cb(self, msg):
...
-    def publish_height(self, z):
-        '''HACK. Langford to fix'''
+    def publish_height(self, msg):
+        '''Sim DVL uses laserscan message to relay height'''
+        print msg
         self.dvl_pub.publish(
             Float64Stamped(
                 header=sub8_utils.make_header(),
-                data=(20.0 + z)
+                data=float(np.mean(msg.ranges))
```
Comment (jpanikulam) [inline simulation/sub8_gazebo/nodes/gazebo_controller.py:62]: Yum

File: simulation/sub8_gazebo/worlds/gripper.world [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0"?>
+  <sdf version="1.4">
```
Comment (jpanikulam) [inline simulation/sub8_gazebo/worlds/gripper.world:2]: What does this world do?

Comment (mattlangford) [inline simulation/sub8_gazebo/worlds/gripper.world:2]: I used it to test the gripper so I didn't have to reload all the sub stuff to make small tweaks

**Comment by mattlangford** [comment]
Also there is a chance that there is one or two added dependencies needed by the ros gazebo tools that are not already included in the install script. If someone tries to run the sim but runs into problems, I will update the install script.

**Pull request merged**